rust/ql/lib/codeql/rust/internal/Definitions.qll

```diff
@@ -166,6 +179,66 @@ private class MethodUse extends Use instanceof NameRef {
   override string getUseType() { result = "method" }
 }
 
+// We don't have entities for the operator symbols, so we approximate a location.
+// The location spans are chosen so that they align with rust-analyzer's jump-to-def
+// behavior in VS Code, which means using weird locations where the end column is
+// before the start column in the case of unary prefix operations.
+private predicate operatorHasLocationInfo(
+  Operation o, string filepath, int startline, int startcolumn, int endline, int endcolumn
+) {
+  o =
+    // `-x`; placing the cursor before `-` jumps to `neg`, placing it after jumps to `x`
+    any(PrefixExpr pe |
+      pe.getLocation().hasLocationInfo(filepath, startline, startcolumn, _, _) and
+      endline = startline and
+      endcolumn = startcolumn - 1
```

Comment:
I think this would fix the backwards-location issue:
```suggestion
      pe.getLocation().hasLocationInfo(filepath, endline, endcolumn + 1, _, _) and
      startline = endline and
      startcolumn = endcolumn
```
Though I'm getting the impression you maybe want the location starting at column 5, ending at column 4 for some reason (which just seems invalid to me).